app/controllers/concerns/mfa_setup_concern.rb

```diff
@@ -6,24 +6,17 @@ def next_setup_path
       auth_method_confirmation_url
     elsif next_setup_choice
       confirmation_path
-    else
-      if user_session[:mfa_selections]
-        analytics.user_registration_mfa_setup_complete(
-          mfa_method_counts: mfa_context.enabled_two_factor_configuration_counts_hash,
-          in_account_creation_flow: user_session[:in_account_creation_flow] || false,
-          enabled_mfa_methods_count: mfa_context.enabled_mfa_methods_count,
-          pii_like_keypaths: [[:mfa_method_counts, :phone]],
-          second_mfa_reminder_conversion: user_session.delete(:second_mfa_reminder_conversion),
-          success: true,
-        )
-      end
+    elsif user_session[:mfa_selections]
+      track_user_registration_mfa_setup_complete_event
       user_session.delete(:mfa_selections)
-      nil
+
+      return sign_up_completed_path
```

Comment:
We can drop the `return` here to use implicit return, which would be consistent with the logic branches above on lines 6 and 8.

```suggestion
      sign_up_completed_path
```